src/sentry/models/outbox.py

```diff
@@ -599,12 +600,19 @@ def process_coalesced(
             # leads to timeouts.
             while True:
                 batch = self.select_coalesced_messages().values_list("id", flat=True)[:100]
-                delete_ids = [item_id for item_id in batch if item_id <= coalesced.id]
+                delete_ids = [item_id for item_id in batch if item_id < coalesced.id]
                 if not len(delete_ids):
                     break
                 self.objects.filter(id__in=delete_ids).delete()
                 deleted_count += len(delete_ids)
 
+            # Only process the highest id after the others have been batch processed.
+            # It's not guaranteed that the ordering of the batch processing is in order,
+            # meaning that failures during deletion could leave an old, staler outbox
+            # alive.
```

Comment:
@corps was this the root cause of the outbox backlog?

Reply:
We're not entirely sure right now what hte outbox backlog was caused by, but suspect org provisioning and or org auth tokens @GabeVillalobos

Reply:
@corps we might need a test case for this specific edge case 😅 doesnt have to be in this PR.